### ISSUE TITLE
comparison of sub-millisecond timestamps

### ISSUE THREAD

**Issue by VladimirAlexiev** [eclipse/rdf4j#4396]
(This is a very minor enhancement request, and if it's hard to implement, I'm fine to close it.)
cc @ mihailradkov

Most people don't have a need to express time to a granularity finer than millisecond.
However, the buildingSmartInternational Data Dictionary (bSDD) has values like
`lastUpdatedUtc='2022-12-21T08:24:16.8080608'`.
XSD allows any number of digits in the decimal part of seconds: https://www.w3.org/TR/xmlschema11-2/#nt-seFrag

Test query (should return `true, "2022-12-21T08:24:16.808002"^^xsd:dateTime`):
```sparql
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
select 
("2022-12-21T08:24:16.808001"^^xsd:dateTime<"2022-12-21T08:24:16.808002"^^xsd:dateTime as ?x) 
("2022-12-21T08:24:16.808001"^^xsd:dateTime+"PT0.000001S"^^xsd:duration as ?y)
{}
```

* Jena handles both addition and comparison ok. Tested at http://sparql.org/sparql.html
* rdf4j handles addition ok, but not comparison. Tested on GraphDB 9.10.3, RDF4J 3.7.3

**Comment by abrokenjester** [comment]
@VladimirAlexiev I can not reproduce the problem in the latest RDF4J (4.2.3). Executing your query on  a test repository (with 
extended evaluation strategy configured to allow for datetime math), I get

```
[x="true"^^<http://www.w3.org/2001/XMLSchema#boolean>;y="2022-12-21T08:24:16.808002"^^<http://www.w3.org/2001/XMLSchema#dateTime>]
```

It's possible that this was fixed since release 3.7.3, or perhaps GraphDB somehow overrides RDF4J default behavior here.

**Comment by abrokenjester** [comment]
Closing for now, but feel free to comment/reopen if I've overlooked something.
